src/crafting.cpp

```diff
@@ -1403,27 +1404,208 @@ void pick_recipes(const inventory &crafting_inv,
+}
+
+void player::make_all_craft(const std::string &id_to_make, int batch_size)
+{
+    make_craft_with_command(id_to_make, batch_size, true);
+}
+
+void player::make_craft_with_command( const std::string &id_to_make, int batch_size, bool is_long )
+{
     const recipe *recipe_to_make = find_recipe( id_to_make );
+
     if( recipe_to_make == nullptr ) {
         return;
     }
-    assign_activity(ACT_CRAFT, recipe_to_make->batch_time(batch_size), recipe_to_make->id);
-    activity.values.push_back( batch_size );
-    last_batch = batch_size;
-    lastrecipe = id_to_make;
+
+    last_craft = craft_command( recipe_to_make, batch_size, is_long, this );
+    last_craft.execute();
 }
 
+template<typename CompType>
+std::string comp_selection<CompType>::nname()
+{
+    switch( use_from ) {
+        case use_from_map:
+            return item::nname( comp.type, comp.count ) + _( " (nearby)" );
+        case use_from_both:
+            return item::nname( comp.type, comp.count ) + _( " (person & nearby)" );
+        default: break;
+    }
 
-void player::make_all_craft(const std::string &id_to_make, int batch_size)
+    return item::nname( comp.type, comp.count );
+}
+
+void craft_command::execute()
 {
-    const recipe *recipe_to_make = find_recipe( id_to_make );
-    if( recipe_to_make == nullptr ) {
+    if( empty() )
         return;
+
+    bool need_selections = true;
+    inventory map_inv;
+    map_inv.form_from_map( crafter->pos3(), PICKUP_RANGE );
+
+    if( has_cached_selections() ) {
+        std::vector<item_selection> missing_items = check_item_components_missing( &map_inv );
+        std::vector<tool_selection> missing_tools = check_tool_components_missing( &map_inv );
+
+        if( missing_items.empty() && missing_tools.empty() ) {
+            need_selections = false; // all items we used previously are still there, so we don't need to do selection
+        } else {
+            /* create a pop up message */
+            std::stringstream ss;
+            ss << "Some components used previously are missing. Continue?";
+
+            if( !missing_items.empty() ) {
+                ss << std::endl << "Item(s): ";
+                for( size_t i = 0; i < missing_items.size(); i++ ) {
+                    if( i != 0 )
+                        ss << ", ";
+                    ss << missing_items[i].nname();
+                }
+            }
+
+            if( !missing_tools.empty() ) {
+                ss << std::endl << "Tool(s): ";
+                for( size_t i = 0; i < missing_tools.size(); i++ ) {
+                    if( i != 0 )
+                        ss << ", ";
+                    ss << missing_tools[i].nname();
+                }
+            }
+
+            std::vector<std::string> options;
+            options.push_back( _( "Yes" ) );
+            options.push_back( _( "No" ) );
+
+            const std::string str = ss.str(); // we NEED a copy
+            int selection = menu_vec( true, _( str.c_str() ), options );
```

Comment:
The translation won't work. The `_` has no idea how to translate the string "Item(s): " (see above) because that string is never marked for translation. Look at `item::info`, specifically the code that lists the recipes one could do with the item. In short: (in this situation) translate every literal string separately right before writing it to stringstream.